Sdk/Speckle.Connectors.Common/Cancellation/CancellationManager.cs

```diff
@@ -27,18 +27,10 @@ private sealed class CancellationItem(CancellationManager manager, string id) :
 
   public int NumberOfOperations => _operationsInProgress.Count;
 
-  /// <summary>
-  /// Get token with registered id.
-  /// </summary>
-  /// <param name="id"> Id of the operation.</param>
-  /// <returns> CancellationToken that belongs to operation.</returns>
-  public CancellationToken GetToken(string id) => _operationsInProgress[id].Token;
+  //if we can't find it then it must be cancelled
+  private CancellationToken GetToken(string id) =>
```

Comment:
why otherwise it generates new token, how it relates to `id` that passed by function argument

Reply:
it's a comment: `//if we can't find it then it must be cancelled`

Doing `new (true)` on CancellationToken struct will mean it's always cancelled.  It could only be missing if previously disposed.  I don't know how the cards are passing an ID that already cancelled but they are.

Reply:
ah, so for later requests we're passing an already cancelled token, which should ensure right behaviour. checking now